src/src/services/models/explorer/createAppModel.ts

```diff
@@ -4956,9 +4992,27 @@ function createAppModel(appConfig: IAppInitialConfig) {
                 suggestions: getSuggestionsByExplorer(appName, data),
               },
             });
+            updateSelectedMetrics();
           });
       }
 
+      function updateSelectedMetrics() {
+        // Check if selected metrics are still available in the select options
+        const selectedMetricsData = model.getState()?.config?.select;
+        if (selectedMetricsData) {
+          const selectedMetrics = selectedMetricsData.options;
+          const selectOptions = model.getState()?.selectFormData?.options;
+          if (selectOptions) {
+            const newSelectedMetrics = selectedMetrics.filter((metric: any) =>
+              selectOptions.find((option: any) => option.key === metric.key),
+            );
+            if (selectedMetrics.length !== newSelectedMetrics.length) {
+              onSelectOptionsChange({ data: newSelectedMetrics, model });
+            }
+          }
+        }
+      }
+
```

Comment:
could this function be shared with the above usage?

Reply:
Fixed by extracting everything into a util function. Might be the best way to proceed with some of the duplicate code that's all over `createAppModel.ts`...